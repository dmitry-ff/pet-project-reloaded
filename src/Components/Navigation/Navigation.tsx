import React from 'react'
import { Link } from 'react-router-dom';
import { NavigationDiv, Logo, Navbar } from './NavigationStyled';
import dog from '../../Images/dog.svg';
import { EmotionJSX } from '@emotion/react/types/jsx-namespace';

const Navigation: React.FC = (): EmotionJSX.Element => {
  return (
    <>
      <NavigationDiv>
        <Link to='/' style={{ textDecoration: 'none', color: 'black' }}>
          <Logo >
            <h2>PET</h2>
            <img src={dog} alt='dog'></img>
            <h2>PROJECT</h2>
          </Logo>
        </Link>
        <Navbar>
          <Link to='Charts'>Charts</Link>
          <Link to='Weather' >Weather</Link>
          <Link to="ReactFeatures">React Features</Link>
        </Navbar>
      </NavigationDiv>
    </>
  );
}

export default Navigation