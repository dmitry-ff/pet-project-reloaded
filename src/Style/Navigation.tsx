import styled from '@emotion/styled';

export const NavigationDiv = styled.div`
    display: flex;
    justify-content: flex-start;
    align-items: center;
    font-size: 14px;
    border-bottom:1px solid rgba(194, 194, 194,0.2);
    box-shadow:0px 0 10px rgba(194, 194, 194,0.4);
    
`
export const Logo = styled.div`
    display:flex;
    margin-left:50px;
    img{
        width:64px;
    }

        text-decoration:none;

`


export const Navbar = styled.nav`
    display:flex;
    margin-left:50px;
        &>a{
        color: rgb(128, 122, 122);
        padding: 25px 20px;
        text-decoration: none;
        font-size: 1.5em;
            &:hover{
                background-color: rgba(238, 237, 237, 0.548);
    }
    }
`

export const FakeElem = styled.div`
    background:transparent;
    padding:20px;
`
