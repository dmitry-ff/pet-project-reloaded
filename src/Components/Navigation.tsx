import React from 'react'
import { Link, Outlet } from 'react-router-dom';
import { NavigationDiv, Logo, Navbar } from '../Style/Navigation';
import dog from '../Images/dog.svg';

function Navigation() {
  return (
    <>
      <NavigationDiv>
        <Link to='/' style={{ textDecoration: 'none', color: 'black' }}>
          <Logo >
            <h2>PET</h2>
            <img src={dog} alt='dog'></img>
            <h2>PROJECT</h2>
          </Logo>
        </Link>
        <Navbar>
          <Link to='Charts'>Charts</Link>
          <Link to='Weather' >Weather</Link>
          <Link to="reactfeatures">React Features</Link>
        </Navbar>
      </NavigationDiv>
      <Outlet />
    </>
  );
}

export default Navigation