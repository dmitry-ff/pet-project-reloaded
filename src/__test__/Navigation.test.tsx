import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Navigation from '../Components/Navigation';
it('renders without crashing', () => {
  shallow(<Navigation />)
})
it("renders correctly", () => {
  const tree = shallow(<Navigation />);
  expect(toJson(tree)).toMatchSnapshot();
});
export { }