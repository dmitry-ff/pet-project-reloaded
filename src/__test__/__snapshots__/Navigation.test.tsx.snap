// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <Styled(div)>
    <Link
      style={
        Object {
          "color": "black",
          "textDecoration": "none",
        }
      }
      to="/"
    >
      <Styled(div)>
        <h2>
          PET
        </h2>
        <img
          alt="dog"
          src="dog.svg"
        />
        <h2>
          PROJECT
        </h2>
      </Styled(div)>
    </Link>
    <Styled(nav)>
      <Link
        to="Charts"
      >
        Charts
      </Link>
      <Link
        to="Weather"
      >
        Weather
      </Link>
      <Link
        to="reactfeatures"
      >
        React Features
      </Link>
    </Styled(nav)>
  </Styled(div)>
  <Outlet />
</Fragment>
`;
